htpasswd.go

```diff
@@ -100,6 +115,23 @@ func (h *HtpasswdFile) throttleTimer() {
 	}
 }
 
+func (h *HtpasswdFile) expiryTimer() {
+	for {
+		time.Sleep(5 * time.Second)
+		now := time.Now()
+		h.mutex.Lock()
+		var zeros [sha256.Size]byte
+		// try to wipe expired cache entries
+		for user, entry := range h.cache {
+			if entry.expiry.After(now) {
+				copy(entry.verifier, zeros[:])
+				delete(h.cache, user)
+			}
+		}
+		h.mutex.Unlock()
+	}
+}
+
```

Comment:
I am not sure I see the advantage of using this method over having a cacheEntryValid function that checks the time on access. It adds the overhead of repeatedly checking the cache for old entries and having another lock on the map, which could slow down accesses. This also adds uncertainty to when the cache will expire because the goroutine has to wake up. At best you will have some cache entries expire 5 seconds late, but depending no how busy other goroutines are, you may end up waiting longer.

Reply:
BTW, ReloadCheck should just move to a goroutine and instead of a time.Sleep pinging a channel, use https://golang.org/pkg/time/#After to do it for you. This would reduce the overhead of calling the ReloadCheck function on each password validation.

Reply:
The point of having this function is the same as zeroing the cache entries in `Reload()`. I'm a bit unsure whether it's worth the trouble to try to wipe old cache entries. But I'd also want to not keep the weaker user:password-hashes in memory permanently.

> This also adds uncertainty to when the cache will expire because the goroutine has to wake up. At best you will have some cache entries expire 5 seconds late, but depending no how busy other goroutines are, you may end up waiting longer.

I don't think it matters whether a cache entry in invalidated after one minute or a bit later on.

> ReloadCheck should just move to a goroutine

Sounds reasonable, but I think that belongs into a separate PR. And I doubt that it makes much of a difference performancewise. At least not when handling a few hundred / thousands of requests per second.